### ISSUE TITLE
superbuild cmake error

### ISSUE THREAD

**Issue by djboersma** [SuperElastix/SimpleElastix#128]
I am trying to install SimpleElastix exactly as advertised, using a build directory that is **not** inside the source directory, but then it looks like the superbuild **does** generate build files inside the source directory, which cmake then subsequently detects and complains about:

```
$ git clone https://github.com/SuperElastix/SimpleElastix
$ mkdir build
$ cd build
$ cmake ../SimpleElastix/SuperBuild/CMakeLists.txt
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
######################################################
# SimpleITK should not be configured & built in the SimpleITK source directory
# You must run cmake in a build directory.
######################################################
CMake Error at /data/scratch/Skandion/forgery/SimpleElastix/CMake/sitkPreventInSourceBuilds.cmake:14 (message):
  Quitting configuration
Call Stack (most recent call first):
  /data/scratch/Skandion/forgery/SimpleElastix/CMake/sitkPreventInSourceBuilds.cmake:18 (AssureOutOfSourceBuilds)
  SuperBuild.cmake:57 (include)
  CMakeLists.txt:38 (include)


-- Configuring incomplete, errors occurred!
See also "/data/scratch/Skandion/forgery/SimpleElastix/SuperBuild/CMakeFiles/CMakeOutput.log".
```

I first thought that maybe my cmake version was buggy (3.5.1) but then I upgraded to the latest version (3.9.3) and still get the same problem. I have spent some time staring at the cmake files to see if I could easily fix it myself but no such luck. Googling didn't produce anything enlightening either, which surprises me a bit, because I'd suspect that I'm not the only user running into this.

After the above command the build directory is empty. There are indeed some new files in the source directory that probably should have landed in the build directory:

```
$ cd ../SimpleElastix
$ git status
On branch master
Your branch is up-to-date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	SuperBuild/CMakeCache.txt
	SuperBuild/CMakeFiles/
	SuperBuild/CTestCustom.cmake

nothing added to commit but untracked files present (use "git add" to track)
```

**Comment by kaspermarstal** [comment]
That is indeed very weird. The only abnormal thing I can spot is that you use `cmake ../SimpleElastix/SuperBuild/CMakeLists.txt` instead of just `cmake cmake ../SimpleElastix/SuperBuild/`. What happens if you use `cmake ../SimpleElastix/SuperBuild/`?

**Comment by djboersma** [comment]
Wow, yes, leaving out `CMakeLists.txt` from the cmake line makes all the difference, it works now.

I have used `cmake` for about a decade, and now that think of it, I normally used it indeed with only the source directory as the main argument (plus -D options). I have no clue why I was suddenly so convinced that I had to pass the path to the top `CMakeLists.txt` file instead. Maybe I'm getting old... Sorry for the noise!
